### PR TITLE
[docker-mgmt-framework]: update mgmt framework docker to support sonic-cli cmd

### PR DESCRIPTION
**Pull request by kwangsuk**
**- Why I did it**
- migrate to python3 support
- add dependent packages for Klish 
- allow login as non-root user

**- How I did it**
update sonic-cli script to start Klish with user name, system name and timeout
update the Dockerfile.j2 to resolve dependent packages
add python3-dev for Klish use 

**- How to verify it**
Incremental buster build with https://github.com/Azure/sonic-mgmt-framework/pull/76 and verify the sonic-cli

**- Which release branch to backport (provide reason below if selected)**
- [ ] 201811
- [ ] 201911
- [ ] 202006

**- Description for the changelog**
Migrate to python3.7 support, update sonic-cli script and resolve package dependencies 

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: dockers/docker-sonic-mgmt-framework/Dockerfile.j2
Comment (lguohan) [inline dockers/docker-sonic-mgmt-framework/Dockerfile.j2:35]: can you remove python3-dev?

**Review by lguohan — CHANGES_REQUESTED** [review]
as comments

File: dockers/docker-sonic-mgmt-framework/Dockerfile.j2
Comment (kwangsuk) [reply dockers/docker-sonic-mgmt-framework/Dockerfile.j2]: Resolved.

**Comment by kwangsuk** [comment]
retest vsimage please

**Comment by sameerdell** [comment]
@lguohan @renukamanavalan @xinanqiu help needed to get this PR merged to be part of 202012 release.

File: dockers/docker-sonic-mgmt-framework/Dockerfile.j2
Comment (renukamanavalan) [inline dockers/docker-sonic-mgmt-framework/Dockerfile.j2:39]: This exposes /etc/passwd & /etc/group inside mgmt FW, which allows/handles Nortbound users' access.
Though this is conditioned upon, INCLUDE_MGMT_FRAMEWORK, this is concerning?

Should not this happen via d-bus ?

Comment (kwangsuk) [reply dockers/docker-sonic-mgmt-framework/Dockerfile.j2]: Those links are not intended in my PR, added accidentally from other branch. It's removed now.

File: dockers/docker-sonic-mgmt-framework/Dockerfile.j2
Comment (renukamanavalan) [inline dockers/docker-sonic-mgmt-framework/Dockerfile.j2:39]: Could you please explain the need for this mapping?

Comment (kwangsuk) [reply dockers/docker-sonic-mgmt-framework/Dockerfile.j2:39]: As of now, the rest server performs basic auth.
 https://github.com/Azure/sonic-mgmt-framework/blob/master/rest/server/pamAuth.go#L117
This will be replaced with JWT token based authentication.

Comment (renukamanavalan) [reply dockers/docker-sonic-mgmt-framework/Dockerfile.j2:39]: I believe d-bus mechanism was the *discussed* solution.

Comment (kwangsuk) [reply dockers/docker-sonic-mgmt-framework/Dockerfile.j2:39]: This PR is limited to start enhanced sonic-CLI and python3 support. I am not aware of the d-bus based solution, but I will ask someone who is familiar with it to have a separate PR to upstream the d-bus based solution.

Comment (kwangsuk) [reply dockers/docker-sonic-mgmt-framework/Dockerfile.j2:39]: @renukamanavalan 
According to the local discussion, it is found that we still need to mount /etc/passwd and /etc/group for rest server to perform group lookups. Once “AAA Improvements” ([HAM](https://github.com/Azure/sonic-buildimage/pull/5553)) is merged to SONiC, it can be obviated since the NSS module will take care of the user/group lookups and there is no need to mount or link the /etc/passwd and /etc/group files anymore. Since the AAA improvements have not been merged, we still need those files accessible in the mgmt-framework. 

As far as security concerned, it should be ok because we are not exposing anything other than what a SONiC user already has access to. Please share your concern if you have.

**Comment by kwangsuk** [comment]
retest baseimage, broadcom, mellanox, vsimage please

**Comment by kwangsuk** [comment]
retest broadcom please

**Comment by kwangsuk** [comment]
retest mellanox please

**Comment by kwangsuk** [comment]
retest vsimage please

**Comment by kwangsuk** [comment]
retest baseimage please

File: dockers/docker-sonic-mgmt-framework/Dockerfile.j2
Comment (renukamanavalan) [reply dockers/docker-sonic-mgmt-framework/Dockerfile.j2:39]: I would like to get a review from @lguohan

Comment (jeff-yin) [reply dockers/docker-sonic-mgmt-framework/Dockerfile.j2:39]: Just to clarify more specifically, see:
https://github.com/martin-belanger/SONiC/blob/aaa_improvements/doc/aaa/AAA%20Improvements/AAA%20Improvements.md#hamd-for-containers

Specifically, with the AAA improvements, the NSS module will take care of the user/group getpwnam lookups and you don’t need to mount or link the /etc/passwd and /etc/group files anymore.

File: dockers/docker-sonic-mgmt-framework/Dockerfile.j2
Comment (jeff-yin) [reply dockers/docker-sonic-mgmt-framework/Dockerfile.j2]: This is discussed in the below thread too, but summarizing it here too:

Currently the /etc/passwd and /etc/group mounting is needed because the REST server is doing group lookups.
https://github.com/Azure/sonic-mgmt-framework/blob/master/rest/server/pamAuth.go#L71 :: IsAdminGroup()

We can eliminate the need to mount /etc/passwd and /etc/group in the container once the “AAA Improvements” (HAM) is merged to SONiC. See:
https://github.com/martin-belanger/SONiC/blob/aaa_improvements/doc/aaa/AAA%20Improvements/AAA%20Improvements.md#hamd-for-containers
Specifically, with the AAA improvements, the NSS module will take care of the user/group lookups and you don’t need to mount or link the /etc/passwd and /etc/group files anymore.

Since the AAA improvements have not been merged, we need those files accessible in the mgmt-framework. I don’t believe that’s a security concern, since you are not exposing anything additional to what a SONiC user already has access to.

**Comment by sameerdell** [comment]
@renukamanavalan @lguohan - AAA improvements (hamd) is work in progress and will not be able to make it in to 202012 release. Until then we have to mount /etc/passwd and /etc/groups. The AAA improvements will remove the need to mount.

Kwan has addressed all the reviews and resolved merge conflicts as well. Requesting review so that we can get this PR merged in 202012 release. Keeping this PR open to long will result again in merge conflicts and too much back & forth.

**Comment by lguohan** [comment]
i think the title can be updated to reflect the pr content.

[docker-mgmt-framework]: update mgmt framework docker to support sonic-cli cmd

**Comment by kwangsuk** [comment]
Yes, Guohan. It is updated accordingly.
Thx/Kwan

**Comment by ben-gale** [comment]
@lguohan, @kwangsuk - next steps pls? It seems like Kwan has addressed Guohan's latest comment above - I still see some failing checks though ....

**Comment by kwangsuk** [comment]
retest mellanox please

**Comment by kwangsuk** [comment]
retest vsimage please

**Comment by ben-gale** [comment]
@lguohan, @kwangsuk  - all checks passing! Let's go .....

**Comment by kwangsuk** [comment]
> as comments

The python3-dev was removed as commented.
`RUN apt-get remove -y g++ python3-dev`

**Comment by sameerdell** [comment]
@lguohan need help with the review of this PR.

**Comment by ben-gale** [comment]
> @lguohan need help with the review of this PR.

@lguohan, @renukamanavalan - yes please!

**Comment by ben-gale** [comment]
@lguohan , @renukamanavalan - please provide your inputs so we can move forwards.

**Comment by ben-gale** [comment]
Note that we believe that this PR is needed for [https://github.com/Azure/sonic-buildimage/issues/6549] - please merge.

File: dockers/docker-sonic-mgmt-framework/Dockerfile.j2
Comment (lguohan) [reply dockers/docker-sonic-mgmt-framework/Dockerfile.j2:39]: can we then just add some comments saying such will be removed when AAA improvements has been merged? do we have AAA improvement pr?

Comment (kwangsuk) [reply dockers/docker-sonic-mgmt-framework/Dockerfile.j2:40]: @lguohan the comment explicitly added as requested, 
The PR for AAA improvements is https://github.com/Azure/sonic-buildimage/pull/5553

**Review by renukamanavalan — APPROVED** [review]

**Pull request merged**
